docs/linking.md

```diff
@@ -0,0 +1,218 @@
+Slang Linking
+=============
+
+The Slang feature around libraries and linking are a *work in progress*. Future versions of Slang are likely to change the API and binary compatibility. Also note that currently reflection is mainly unsupported. 
+
+In many languages it is possible to compile source files into binaries such as libraries and object files. Later these files can be combined perhaps with other source files to produce a result such as an executable. Slang now has experimental support for such a feature. 
+
+To make such a feature work we need a few abilities
+
+* To create a library from source 
+* A way for symbols to be exported and imported
+* Linkage - ability to combine source/libraries to produce things (such as other libraries, executables, etc)
+* A way to specify libraries to be used in linkage
+
+Libraries
+---------
+
+A library can be created from source from the Slang command line with the `-o` option naming the output file. The filename currently requires a `slang-lib` or `slang-module` extension. This extension will mean the library will be stored as serialized Slang IR. The option `-no-codegen` is also typically specified because we don't want to generate any *target* code, such as dxil, spir-v and so forth when producing a library. 
```

Comment:
I agree on the `-no-codegen` thing as it stands for now, but it is entirely likely that the container format will be extended to support storing the generated DXBC/DXIL/SPIR-V code for entry points that can be code-generated.

Hmm... I guess if we are strictly trying to follow the flow of how traditional compilation works, then we'd need to draw a distinction between object code and binary code, and only store kernels in the latter. I'm imagining the container format wouldn't care about that distinction (just as ELF doesn't care), but we'd need some policy about which extensions represent object code and which represent binaries, if we go this route.

The other option is the more Java/C# route where there is no strong distinction between object and binary files...